src/AutofacSerilogIntegration/ContextualLoggingModule.cs

```diff
@@ -48,8 +50,17 @@ protected override void Load(ContainerBuilder builder)
 
                 return logger;
             })
-            .As<ILogger>()
-            .ExternallyOwned();
+            .As<ILogger>();
+
+            if (_dispose)
+            {
+                registration.SingleInstance()
```

Comment:
> The problem though, is that the container needs to return a different instance of ILogger depending on the type it's being injected into.

That's exactly the behavior I am observing. For the purpose of this test I am injecting an ILogger into 3 distinct API controllers. All 3 controllers receive their distinct instances of ILogger enriched with a FixedPropertyEnricher containing the SourceContext property set to their type name.
Courtesy of:
```c#
            registration.Preparing += (sender, args) =>
            {
                var log = args.Context.Resolve<ILogger>().ForContext(registration.Activator.LimitType);
                args.Parameters = new[] {TypedParameter.From(log)}.Concat(args.Parameters);
            };
```
[source](https://github.com/nblumhardt/autofac-serilog-integration/blob/cbea2737f79b35e2ab332e4cd167d4ebadc0ca0d/src/AutofacSerilogIntegration/ContextualLoggingModule.cs#L93-L97
)